### PR TITLE
fix(yaml-cfn): do not deserialize year-month-date as strings

### LINKED ISSUE #13709: Issue importing with CfnInclude when template do not have quotes
I encounter this issue when trying to import CloudFormation YAML Template to create IAM role.

The template have the following header:
AWSTemplateFormatVersion: 2010-09-09
and also version with date
Version: 2012-10-17.

The error I encounter is when trying to deploy the template to the AWS via CDK.
The build is fine but when deploying I encounter the following error:
failed: Error [ValidationError]: Template format error: unsupported value for AWSTemplateFormatVersion.

I noticed that the same template can be deploy via AWS CLI and validated without issue using:
aws cloudformation create-stack --stack-name
and 
aws cloudformation validate-template --template-body

I think this is a bug because AWS CLI have no issue with this why would deploying via CDK raises error.

### Reproduction Steps

Deploying via CDK with any template with the above header would work that do not have quotes around the date. 

### What did you expect to happen?

it would throw an error 
failed: Error [ValidationError]: Template format error: unsupported value for AWSTemplateFormatVersion.
### What actually happened?

failed: Error [ValidationError]: Template format error: unsupported value for AWSTemplateFormatVersion.

### Environment

  - **CDK CLI Version  :1.81.0**
  - **Framework Version:**
  - **Node.js Version:** 
  - **OS               : MacOS**
  - **Language (Version): Typescript** 

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by skinny85**
Currently, we are using the `yaml-1.1` schema when de-serializing YAML documents,
Unfortunately, this has the side effect of treating unquoted parts of the template like '2010-09-09'
as Date objects, instead of just simple strings.
This has been noted by a customer using the cloudformation-include module,
but I assume a very similar problem exists for other places we parse YAML,
like cloudformation-diff.

Switch to the `core` schema from `yaml-1.1`,
where those are treated as strings, instead of dates.

Fixes #13709

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/yaml-cfn/lib/yaml.ts [hunk 1]
```diff
@@ -54,6 +54,6 @@ const shortForms: yaml_types.Schema.CustomTag[] = [
 function parseYamlStrWithCfnTags(text: string): any {
   return yaml.parse(text, {
     customTags: shortForms,
-    schema: 'yaml-1.1',
+    schema: 'core',
```
Comment (RomainMuller) [inline packages/@aws-cdk/yaml-cfn/lib/yaml.ts:57]: But are we sure using the `core` schema is not going to break _other_ stuff? I.e: is `"No"` going to be encoded correctly (read: quoted), or is it going to be encoded YAML-1.2-style (which in Yaml 1.1 is an alias for `false`?

Comment (skinny85) [reply packages/@aws-cdk/yaml-cfn/lib/yaml.ts:57]: Good point! I added a test confirming that `No` is deserialized correctly (as `"No"`, not `false`).

Let me know if you want to see some more tests of some edge-cases in the YAML handling!

**Review by RomainMuller — APPROVED** [review]
